community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStore.java

```diff
@@ -1020,4 +1041,35 @@ public boolean visit( PageCursor element ) throws IOException
             }
         } );
     }
+
+    private static class HighestCommittedTransaction
```

Comment:
This abstraction incurs some overhead, one more object for every call to `transactionCommitted` and `getLastCommittedTx`. It's only internal to this class and we can make accessors for a `long[2]` to make it very readable and maintainable.